CMakeLists.txt

```diff
@@ -114,8 +114,12 @@ if (NOT DEFINED QL_BOOST_COMPONENTS)
     # Boost thread and unit_test_framework needed for tests and benchmark
     if (QL_BUILD_TEST_SUITE OR QL_BUILD_BENCHMARK)
         set(QL_BOOST_COMPONENTS thread unit_test_framework)
-    else()
+    # Boost thread needed for sessions, singleton thread-safe init adn thread-safe observer pattern
+    elseif(QL_ENABLE_SESSIONS OR QL_ENABLE_SINGLETON_THREAD_SAFE_INIT OR QL_ENABLE_THREAD_SAFE_OBSERVER_PATTERN)
         set(QL_BOOST_COMPONENTS thread)
+    # Boost headers only needed for all other configurations
+    else()
+        set(QL_BOOST_COMPONENTS "")
     endif()
```

Comment:
In the default configuration tests don't need thread either.  It should be something like:
- start from empty;
- if tests or benchmark, add unit_test_framework;
- if any of those configs, add thread

The last two are independent, not an if/else.

Reply:
Ok, done. I mistakenly thought that unit_test_framework had a dependency on thread.